enterprise_access/apps/bffs/context.py

```diff
@@ -136,158 +143,44 @@ def _initialize_enterprise_customer_users(self):
+                self.request,
                 traverse_pagination=True
             )
-        except Exception as e:  # pylint: disable=broad-except
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.exception('Error retrieving linked enterprise customers')
             self.add_error(
                 user_message='Error retrieving linked enterprise customers',
-                developer_message=str(e)
+                developer_message=f'Could not fetch enterprise customer users. Error: {exc}'
             )
             return
 
         # Set enterprise features from the response
         self._enterprise_features = enterprise_customer_users_data.get('enterprise_features', {})
 
-        # Parse the enterprise customer user data
-        enterprise_customer_users = enterprise_customer_users_data.get('results', [])
-        active_enterprise_customer = self._get_active_enterprise_customer(enterprise_customer_users)
-        enterprise_customer_user_for_requested_customer = next(
-            (
-                enterprise_customer_user
-                for enterprise_customer_user in enterprise_customer_users
-                if self._enterprise_customer_matches_slug_or_uuid(enterprise_customer_user.get('enterprise_customer'))
-            ),
-            None
-        )
-
-        # If no enterprise customer user is found for the requested customer (i.e., request user not explicitly
-        # linked), but the request user is staff, attempt to retrieve enterprise customer metadata from the
-        # `/enterprise-customer/` LMS API endpoint instead.
-        if not enterprise_customer_user_for_requested_customer:
-            staff_enterprise_customer = self._get_staff_enterprise_customer()
-        else:
-            staff_enterprise_customer = None
-
-        # Determine the enterprise customer user to display
-        requested_enterprise_customer = (
-            enterprise_customer_user_for_requested_customer.get('enterprise_customer')
-            if enterprise_customer_user_for_requested_customer else None
-        )
-        enterprise_customer = self._determine_enterprise_customer_for_display(
-            active_enterprise_customer=active_enterprise_customer,
-            requested_enterprise_customer=requested_enterprise_customer,
-            staff_enterprise_customer=staff_enterprise_customer,
-        )
-
-        # Update the context data with the enterprise customer user information
+        # Parse/transform the enterprise customer users data and update the context data
+        transformed_data = {}
+        try:
+            transformed_data = transform_enterprise_customer_users_data(
+                enterprise_customer_users_data,
+                request=self.request,
+                enterprise_customer_slug=self.enterprise_customer_slug,
+                enterprise_customer_uuid=self.enterprise_customer_uuid,
+            )
```

Comment:
Follow-up from our discussion about this offline: while `enterpriseFeatures` is added to `HandlerContext` above, it's never exposed in the BFF response. This will be added as a follow-up PR.

Reply:
For posterity, the `enterpriseFeatures` and `enterpriseCustomer` attributes are added in this PR https://github.com/openedx/enterprise-access/pull/597